ivy/functional/frontends/torch/nn/functional/loss_functions.py

```diff
@@ -134,3 +134,32 @@ def smooth_l1_loss(
     ret = reduction(loss)
 
     return ret
+
+
+def nll_loss(
+        input, target,
+        weight=None,
+        size_average=None,
+        ignore_index=- 100,
+        reduce=None,
+        reduction='mean'
+):
+    probs = ivy.exp(input)
+    depth = input.shape[-1]
+    one_hot_target = ivy.one_hot(target, depth)
+    ret = ivy.cross_entropy(one_hot_target, probs, reduction='none', epsilon=0.0)
+    if weight is not None:
```

Comment:
handle the strange way in which pytorch applies mean reduction when we pass weights as argument, so we cannot apply directly the reduction in this case.  See [Pytorch docs](https://pytorch.org/docs/stable/generated/torch.nn.NLLLoss.html)